src/Service/FixtureService.php

```diff
@@ -28,49 +33,85 @@
  */
 class FixtureService
 {
+    /**
+     * @var Loader
+     */
     private $loader;
+
+    /**
+     * @var bool
+     */
     private $autoload;
+
+    /**
+     * @var array
+     */
     private $fixtures;
+
+    /**
+     * @var array
+     */
     private $directories;
-    private $migrations;
+
+    /**
+     * @var Kernel
+     */
     private $kernel;
+
+    /**
+     * @var EntityManagerInterface
+     */
     private $entityManager;
+
+    /**
+     * @var PlatformListener
+     */
     private $listener;
-    private $useBackup;
+
+    /**
+     * @var null|BackupService
+     */
     private $backupService;
 
     /**
-     * @var \Doctrine\Common\DataFixtures\ProxyReferenceRepository
+     * @var null|ProxyReferenceRepository
      */
     private $referenceRepository;
 
     /**
-     * Constructor.
+     * FixtureService constructor.
      *
-     * @param \Symfony\Component\DependencyInjection\ContainerInterface $container Service container
-     * @param \Symfony\Component\HttpKernel\Kernel                      $kernel    Application kernel
+     * @param Kernel $kernel
+     * @param bool   $autoload
+     * @param array  $fixtures
+     * @param array  $directories
      */
-    public function __construct(ContainerInterface $container, Kernel $kernel)
+    public function __construct(Kernel $kernel, bool $autoload, array $fixtures, array $directories)
```

Comment:
Do we really need `Kernel` here?

Reply:
Yes. I see no other way to list the bundles, get the cache dir or access on the doctrine fixtures loader.

Reply:
I think the bundle list should be deprecated in favour of fixture services, since the hardcoded subdir is not a good option anyway

Reply:
After that cache dir and loader can be injected directly, making this service a bit more decoupled